protos/mission_fixed_wing/mission_fixed_wing.proto

```diff
@@ -0,0 +1,243 @@
+}
+message SetCurrentMissionItemResponse {
+    MissionResult mission_result = 1;
+}
+
+message IsMissionFinishedRequest {}
+message IsMissionFinishedResponse {
+    MissionResult mission_result = 1;
+    bool is_finished = 2; // True if the mission is finished and the last mission item has been reached
+}
+
+message SubscribeMissionProgressRequest {}
+message MissionProgressResponse {
+    MissionProgress mission_progress = 1; // Mission progress
+}
+
+message GetLandAfterMissionRequest {}
+message GetLandAfterMissionResponse {
+    MissionResult mission_result = 1;
+    bool enable = 2; // If true, trigger a landing at the end of the mission
+}
+
+message SetLandAfterMissionRequest {
+    double land_lat = 1; // Land latitude (in degrees)
+    double land_lng = 2; // Land longitude (in degrees)
+    float land_alt = 3; // Land altitude AMSL (in meters)
+    float loiter_alt = 4; // Loiter altitude AMSL (in meters)
+    bool loiter_cw = 5; // Loiter clockwise
+    float glide_slope = 6; // Glide slope in degrees. Loiter coords are calculated from this
+}
+message SetLandAfterMissionResponse {
+    MissionResult mission_result = 1;
+}
+
+message ImportQgroundcontrolMissionRequest {
+    string qgc_plan_path = 1; // File path of the QGC plan
+}
+message ImportQgroundcontrolMissionResponse {
+    MissionResult mission_result = 1;
+    MissionPlan mission_plan = 2; // The mission plan
+}
+
+/*
+ * Type representing a mission item.
+ *
+ * A MissionItem can contain a position and/or actions.
+ * Mission items are building blocks to assemble a mission,
+ * which can be sent to (or received from) a system.
+ * They cannot be used independently.
+ */
+message MissionItem {
+    double latitude_deg = 1 [(mavsdk.options.default_value)="NaN", (mavsdk.options.epsilon)=0.0000001]; // Latitude in degrees (range: -90 to +90)
+    double longitude_deg = 2 [(mavsdk.options.default_value)="NaN", (mavsdk.options.epsilon)=0.0000001]; // Longitude in degrees (range: -180 to +180)
+    float relative_altitude_m = 3 [(mavsdk.options.default_value)="NaN"]; // Altitude relative to takeoff altitude in metres
+    float speed_m_s = 4 [(mavsdk.options.default_value)="NaN"]; // Speed to use after this mission item (in metres/second)
+    bool is_fly_through = 5 [(mavsdk.options.default_value)="false"]; // True will make the fixed-wing drone fly through without stopping, while false will make the fixed-wing drone hold around the waypoint
+    float gimbal_pitch_deg = 6 [(mavsdk.options.default_value)="NaN"]; // Gimbal pitch (in degrees)
+    float gimbal_yaw_deg = 7 [(mavsdk.options.default_value)="NaN"]; // Gimbal yaw (in degrees)
+    CameraAction camera_action = 8; // Camera action to trigger at this mission item
+    float loiter_time_s = 9 [(mavsdk.options.default_value)="NaN"]; // Loiter time (in seconds)
+    double camera_photo_interval_s = 10 [(mavsdk.options.default_value)="1.0"]; // Camera photo interval to use after this mission item (in seconds)
```

Comment:
So we probably need an enum here for takeoff and landing, something like `FixedwingTakeoff`, `FixedwingApproach`, `FixedwingLanding`, right?

Reply:
Is clumping those 3 into the 1 `MissionItem` the right way of doing it?

Reply:
How else would you add it? I'm open for suggestions.

Reply:
We could have a MissionPlan consisting of a takeoff part, flight part, and land part, for instance...

Reply:
Well for a valid mission you need a landing part. Take off is only need if you're grounded. So maybe have it such that `optional takeoff creator` -> `mission planning as it currently is` -> `mandatory landing creator`.  

I was just tinkering about with how to construct a valid fixed-wing mission and this seems to be the bare useful minimum that's allowed by PX4. 

```python
#!/usr/bin/env python3

import asyncio
from mavsdk import System, Mission_raw

async def run():
    drone = System()
    await drone.connect(system_address="udp://:14540")

    print("Waiting for drone to connect...")
    async for state in drone.core.connection_state():
        if state.is_connected:
            print(f"Drone discovered with UUID: {state.uuid}")
            break
    # Landing with a 6deg slope calcualted by QGC
    items = []
    items.append(
        mission_raw.MissionItem( # MAV_CMD_DO_LAND_START 189
            0,2,189,1,1,0,0,0,0,0,0,0,0)
    )
    items.append( # MAV_CMD_NAV_LOITER_TO_ALT 31
        mission_raw.MissionItem(
            1,6,31,0,1,1,75,0,1,558683815,-42929235,40.0,0)
    )
    items.append( # MAV_CMD_NAV_LAND 21
        mission_raw.MissionItem(
            2,6,21,0,1,0,0,0,0,558688774,-42867694,0,0
        )
    )
    await drone.mission_raw.upload_mission(items)
    await drone.mission_raw.start_mission()

if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    loop.run_until_complete(run())
```